mypy/type_visitor.py

```diff
@@ -253,7 +253,7 @@ class TypeQuery(SyntheticTypeVisitor[T]):
 
     def __init__(self, strategy: Callable[[Iterable[T]], T]) -> None:
         self.strategy = strategy
-        self.seen = []  # type: List[Type]
+        self.seen_aliases = set()  # type: Set[TypeAliasType]
```

Comment:
Whill we ever need to clear this? If the visitor is single-use, mention this in the docstring.